internal/cfgplugins/bgp.go

```diff
@@ -622,3 +624,112 @@ func ConfigureBGPNeighbor(t *testing.T, dut *ondatra.DUTDevice, ni *oc.NetworkIn
 	nAfiSafi.GetOrCreateAddPaths().Send = ygot.Bool(sendReceivePaths)
 }
+
+// ConfigureDUTBGP configures BGP on the DUT using OpenConfig.
+func ConfigureDUTBGP(t *testing.T, dut *ondatra.DUTDevice, dutAS, ate1AS, ate2AS, ate3AS, ate4AS, ate5AS, ecmpMaxPath uint32,
+	dutP1, dutP2, dutP7, ateP1, ateP2, ateP7, dutLag1, dutLag2, ateLag1, ateLag2 attrs.Attributes) {
+	t.Helper()
+	d := gnmi.OC()
+
+	dutBgpConfPath := d.NetworkInstance(deviations.DefaultNetworkInstance(dut)).Protocol(
+		oc.PolicyTypes_INSTALL_PROTOCOL_TYPE_BGP, "BGP",
+	)
+
+	// Create BGP config
+	dutBgpConf := &oc.NetworkInstance_Protocol{
+		Identifier: oc.PolicyTypes_INSTALL_PROTOCOL_TYPE_BGP,
+		Name:       ygot.String("BGP"),
+		Bgp:        &oc.NetworkInstance_Protocol_Bgp{},
+	}
+
+	bgp := dutBgpConf.Bgp
+	global := bgp.GetOrCreateGlobal()
+	global.As = ygot.Uint32(dutAS)
+	global.RouterId = ygot.String(dutP1.IPv4)
+
+	af4 := global.GetOrCreateAfiSafi(oc.BgpTypes_AFI_SAFI_TYPE_IPV4_UNICAST)
+	af4.Enabled = ygot.Bool(true)
+	af6 := global.GetOrCreateAfiSafi(oc.BgpTypes_AFI_SAFI_TYPE_IPV6_UNICAST)
+	af6.Enabled = ygot.Bool(true)
+
+	// Append BGP neighbors
+	appendBGPNeighbor(t, bgp, ate1AS, dutP1.Name, ateP1.IPv4, ateP1.IPv6, false)
+	appendBGPNeighbor(t, bgp, ate5AS, dutP7.Name, ateP7.IPv4, ateP7.IPv6, false)
+	appendBGPNeighbor(t, bgp, ate2AS, dutP2.Name, ateP2.IPv4, ateP2.IPv6, false)
+	appendBGPNeighbor(t, bgp, ate3AS, dutLag1.Name, ateLag1.IPv4, ateLag1.IPv6, true)
+	appendBGPNeighbor(t, bgp, ate4AS, dutLag2.Name, ateLag2.IPv4, ateLag2.IPv6, true)
+
+	// Apply config
+	gnmi.Replace(t, dut, dutBgpConfPath.Config(), dutBgpConf)
+
+	// Handle multipath deviation
+	if deviations.MultipathUnsupportedNeighborOrAfisafi(dut) {
+		t.Log("Executing CLI commands for multipath deviation")
+		bgpRouteConfig := fmt.Sprintf(`		
+		router bgp %d
+		address-family ipv4
+		maximum-paths %[2]d ecmp %[2]d
+		bgp bestpath as-path multipath-relax
+		address-family ipv6
+		maximum-paths %[2]d ecmp %[2]d
+		bgp bestpath as-path multipath-relax
+		`, dutAS, ecmpMaxPath)
+		helpers.GnmiCLIConfig(t, dut, bgpRouteConfig)
+	} else {
+		// TODO: Once multipath is fully supported via OpenConfig across all platforms,
+		// remove CLI fallback and rely solely on OC configuration.
+		af4.GetOrCreateUseMultiplePaths().Enabled = ygot.Bool(true)
+		af4.GetOrCreateUseMultiplePaths().GetOrCreateEbgp().AllowMultipleAs = ygot.Bool(true)
+		af6.GetOrCreateUseMultiplePaths().Enabled = ygot.Bool(true)
+		af6.GetOrCreateUseMultiplePaths().GetOrCreateEbgp().AllowMultipleAs = ygot.Bool(true)
+		af4.GetOrCreateUseMultiplePaths().GetOrCreateIbgp().SetMaximumPaths(ecmpMaxPath)
+		af6.GetOrCreateUseMultiplePaths().GetOrCreateIbgp().SetMaximumPaths(ecmpMaxPath)
+	}
+}
+
+// appendBGPNeighbor appends IPv4 and IPv6 neighbors (and their peer groups) to the BGP configuration.
+func appendBGPNeighbor(t *testing.T, bgp *oc.NetworkInstance_Protocol_Bgp, ateAs uint32, portName, neighborIpV4, neighborIpV6 string, isLag bool) {
```

Comment:
This should be exported (ie: capitalized) so _test.go files can import and call this function.

As with other exported functions, the parameters should be a struct